src/app/api/Mock/MockApi.ts

```diff
@@ -271,18 +322,19 @@ export default class MockClient implements API {
         return shiftToAdd as Shift;
     }
 
-    async deleteShift(shiftId: IdType): Promise<void> {
+    async deleteShift(shiftIds: IdType[]): Promise<void> {
         await randomDelay();
-        if (shiftId == null) {
+        if (!shiftIds) {
             throw new Error('No ID specified');
         }
 
-        const shiftIndex = sheriffShifts.findIndex((shift) => shift.id === shiftId);
-        if (shiftIndex < 0) {
-            throw Error(`No shift could be located for ${shiftId}`);
-        }
-
-        sheriffShifts.splice(shiftIndex, 1);
+        shiftIds.forEach(shiftId => {
```

Comment:
here if you wanted to you could use map:
```ts
shiftIds.map(id=>sheriffShifts.findIndex(s=>s.id==id))
.filter(index=>index>=0) // remove indices < 0
.foreach(index=>sheriffShifts.splice(index,1))  // for each index splice the sheriffShifts
```

Doesn't need to be changed just showing you a more functional way of doing the same thing.